### PR TITLE
Remove an unused variable in _subclasses.fake_impls

### PR DESCRIPTION
**Pull request by rec**
* Extracted from https://github.com/pytorch/pytorch/pull/133492

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #13808

### CODE REVIEW

File: torch/_subclasses/fake_impls.py [hunk 1]
```diff
@@ -630,7 +630,7 @@ def multi_device_op_default(fake_mode, func, *args, **kwargs):
 @register_op_impl(aten.slice_scatter.out)
 def multi_device_op_out(fake_mode, func, *args, **kwargs):
     with in_kernel_invocation_manager(fake_mode):
-        out = func(*args, **kwargs)
+        func(*args, **kwargs)
 
     _, new_kwargs = normalize_function(
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
```
Comment (Skylion007) [inline torch/_subclasses/fake_impls.py:636]: Is the above part a bug? The function is called again right below with `normalize_function`?

Comment (Skylion007) [reply torch/_subclasses/fake_impls.py:636]: Oh wait, nroamlize function doesn't actuall call it?

Comment (rec) [reply torch/_subclasses/fake_impls.py:636]: It seems `normalize function` takes a possible function call and normalizes the arguments, but doesn't call it.

Comment (Skylion007) [reply torch/_subclasses/fake_impls.py:636]: Do we actually need to call it though? That's what I am confused by

**Review by albanD — APPROVED** [review]
This is an out variant function meaning that the change is done inplace into the "out" kwargs. Sounds like the right fix to me

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**